functional-tests/src/test/groovy/specs/app/LoginSpec.groovy

```diff
@@ -42,4 +31,116 @@ class LoginSpec extends FixtureSpec {
       ClickLink           || AssertPage
       "ForgotPasswordBtn" || ForgotPasswordPage
   }
+
+  def "Navigate Page from: LoginPage, log in as Admin, Assert Page: HomePage"() {
+
+    given: "I start on the LoginPage, am not logged in, and have an account"
+      fixture_files =['admin_user_fixture']
+      fixture(FixtureSpec.INSERT)
+      to LoginPage
+    when: "and I log in with valid credentials"
+      login(_username, _password)
+    then: "I am logged in and returned to the HomePage"
+      at HomePage
+      !footerModule.LogInSiteminderLink
+      footerModule.LogOutLink
+    cleanup:
+      fixture(FixtureSpec.REMOVE)
+  }
+
+  def "Navigate Page from: LoginPage, attempt to login using invalid usename and invalid password, Assert Page: SMErrorPage"() {
+
+    given: "I start on the LoginPage, am not logged in"
+      fixture_files =['admin_user_fixture']
+      fixture(FixtureSpec.INSERT)
+      to LoginPage
+    when: "and attempt to log in with an invalid username and invalid password"
+      login('foo', 'bar')
+    then: "I am redirected to the error page, am not logged in, and am provided an unauthorized message"
+      at SMErrPage
+
+      Bceid_business_authentication_failure_message
+      !Bceid_business_authentication_failure_message.displayed
+
+      Bceid_user_authentication_failure_message
+      !Bceid_user_authentication_failure_message.displayed
+
+      Bceid_individual_authentication_failure_message
+      !Bceid_individual_authentication_failure_message.displayed
+
+      Idir_user_authentication_failure_message
+      !Idir_user_authentication_failure_message.displayed
+
+      Unknown_user_type_authentication_failure_message
+      Unknown_user_type_authentication_failure_message.displayed
+
```

Comment:
It would be worth asserting that the error message in these fields matches what we expect.
ie: Unknown_user_type_authentication_failure_message.text().equals("user is not known to ESM or password invalid")

Reply:
IMO this is similar to the title comment above --- it makes the test too brittle